packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -1174,6 +1204,276 @@ contract OPContractsManagerDeployer is OPContractsManagerBase {
+        // owner and therefore have the same access models.
+        address proxyAdminOwner = _input.opChainConfigs[0].proxyAdmin.owner();
+
+        // Deploy the new ETHLockbox.
+        // NOTE that here and in the rest of this function we use block.timestamp as a fake chain
+        // id for any new contracts that are deployed. The L2 chain id is not used for anything
+        // except the salt mixer, so making it the same as the block timestamp means that new
+        // contracts will be generated every time this function is called. This is totally fine for
+        // our purposes here, there's no strong need to have deterministic addresses ahead of time.
+        IETHLockbox newEthLockbox = IETHLockbox(
+            deployProxy({
+                _l2ChainId: block.timestamp,
+                _proxyAdmin: _input.opChainConfigs[0].proxyAdmin,
+                _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                _contractName: "ETHLockbox-Interop"
+            })
+        );
+
+        // Initialize the new ETHLockbox.
+        // Note that this authorizes the portals to use the ETHLockbox.
+        upgradeToAndCall(
+            _input.opChainConfigs[0].proxyAdmin,
+            address(newEthLockbox),
+            getImplementations().ethLockboxImpl,
+            abi.encodeCall(IETHLockbox.initialize, (portals[0].superchainConfig(), portals))
+        );
+
+        // Deploy the new DisputeGameFactory.
+        IDisputeGameFactory newDisputeGameFactory = IDisputeGameFactory(
+            deployProxy({
+                _l2ChainId: block.timestamp,
+                _proxyAdmin: _input.opChainConfigs[0].proxyAdmin,
+                _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                _contractName: "DisputeGameFactory-Interop"
+            })
+        );
+
+        // Initialize the new DisputeGameFactory.
+        upgradeToAndCall(
+            _input.opChainConfigs[0].proxyAdmin,
+            address(newDisputeGameFactory),
+            getImplementations().disputeGameFactoryImpl,
+            abi.encodeCall(IDisputeGameFactory.initialize, (proxyAdminOwner))
+        );
+
+        // Deploy the new AnchorStateRegistry.
+        IAnchorStateRegistry newAnchorStateRegistry = IAnchorStateRegistry(
+            deployProxy({
+                _l2ChainId: block.timestamp,
+                _proxyAdmin: _input.opChainConfigs[0].proxyAdmin,
+                _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                _contractName: "AnchorStateRegistry-Interop"
+            })
+        );
+
+        // Select the correct game type based on the type being used by the existing portals.
+        GameType newGameType;
+        if (_input.usePermissionlessGame) {
+            newGameType = GameTypes.SUPER_CANNON;
+        } else {
+            newGameType = GameTypes.SUPER_PERMISSIONED_CANNON;
+        }
```

Comment:
it's not clear to me how the game type chosen relates to the "existing portals"

Reply:
Outdated comment, need to fix